### PR TITLE
build cmake for linux-armv7

### PR DESCRIPTION
**Pull request by pborcin**
CMake is not prebuilt for armv7, developers need to build it themselves and upload it manually.
This PR automates the task.

### CODE REVIEW

**Comment by pborcin** [comment]
@tomassebestik PTAL

**Comment by tomassebestik** [comment]
@JurajSadel PTAL

File: .github/workflows/build-cmake-linux-armv7.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+on:
+  workflow_dispatch:
+    inputs:
+      cmake_version:
+        description: >
+          Cmake version to build and upload
+          For example 3.23.1
+        type: string
+        required: true
+        default: 'master'
```
Comment (tomassebestik) [inline .github/workflows/build-cmake-linux-armv7.yml:12]: What is this line (`default: 'master'`) used for?

**Comment by tomassebestik** [comment]
@pborcin LGTM

File: .github/workflows/build-cmake-linux-armv7.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - self-hosted
+      - ARM
+    container:
+      image: ghcr.io/espressif/github-esp-dockerfiles/pyenv_rust_powershell:v1
+      options: --privileged
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Download and unpack cmake
+        shell: bash
```
Comment (JurajSadel) [inline .github/workflows/build-cmake-linux-armv7.yml:26]: We don't need all of these"shell: bash" but we also have it in other workflows ...

File: .github/workflows/build-cmake-linux-armv7.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        working-directory: 'cmake-${{ inputs.cmake_version }}/cmake-build'
+        run: cpack -G TGZ && cpack -G STGZ
+      - name: Upload cmake to s3
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
+          AWS_BUCKET: ${{ secrets.DL_BUCKET }}
+          PREFIX: 'dl/cmake'
```
Comment (JurajSadel) [inline .github/workflows/build-cmake-linux-armv7.yml:47]: `PREFIX` is not being used anywhere

File: .github/workflows/build-cmake-linux-armv7.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        shell: bash
+        working-directory: 'cmake-${{ inputs.cmake_version }}/cmake-build'
+        run: |
```
Comment (JurajSadel) [inline .github/workflows/build-cmake-linux-armv7.yml:50]: We don't need `|` here

**Review by JurajSadel — APPROVED** [review]
I left a few nitpicking comments, I haven't tested the actual workflow on RPi.
LGTM

**Comment by tomassebestik** [comment]
Tested, looks great: https://dl.espressif.com/dl/cmake/cmake-3.26.1-Linux-armv7l.tar.gz

Thank you @pborcin !

**Pull request merged**